controllers/storagecluster/onboarding_validation_keys_generator_job.go

```diff
@@ -0,0 +1,114 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+)
+
+const (
+	providerAPIServerImage              = "PROVIDER_API_SERVER_IMAGE"
+	onboardingSecretGeneratorImage      = "ONBOARDING_SECRET_GENERATOR_IMAGE"
+	onboardingJobName                   = "onboarding-secret-generator"
+	onboardingTicketPublicKeySecretName = "onboarding-ticket-key"
+)
+
+type onboardingValidationKeysGeneratorJob struct{}
+
+func (o *onboardingValidationKeysGeneratorJob) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	if instance.Spec.AllowRemoteStorageConsumers {
+		r.Log.Info("onboarding validation key generator job: Spec.AllowRemoteStorageConsumers is enabled")
+		if res, err := o.createJob(r, instance); err != nil {
+			return reconcile.Result{}, err
+		} else if !res.IsZero() {
+			return res, nil
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (o *onboardingValidationKeysGeneratorJob) ensureDeleted(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	onboardingSecretGeneratorJob := getOnboardingJobObject(instance)
+	err := r.Client.Delete(context.TODO(), onboardingSecretGeneratorJob)
+
+	if err != nil && errors.IsNotFound(err) {
+		r.Log.Info("Failed to delete resource", "Kind", onboardingSecretGeneratorJob.GetObjectKind(), "Name", onboardingSecretGeneratorJob.GetName())
+		return reconcile.Result{}, nil
+	}
+
+	return reconcile.Result{}, err
+}
+
+func getOnboardingJobObject(instance *ocsv1.StorageCluster) *batchv1.Job {
+
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      onboardingJobName,
+			Namespace: instance.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			// Eligible to delete automatically when job finishes
+			TTLSecondsAfterFinished: ptr.To(int32(0)),
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy:      corev1.RestartPolicyOnFailure,
+					ServiceAccountName: onboardingJobName,
+					Containers: []corev1.Container{
+						{
+							Name:    onboardingJobName,
+							Image:   os.Getenv(onboardingSecretGeneratorImage),
+							Command: []string{"/usr/local/bin/onboarding-secret-generator"},
+							Env: []corev1.EnvVar{
+								{
+									Name:  util.OperatorNamespaceEnvVar,
+									Value: os.Getenv(util.OperatorNamespaceEnvVar),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
```

Comment:
Why do we need a function for this? 
Let's have the fixed struct as a var at the package scope (at the top of the file)

Reply:
In general this is an established pattern in ocs-operator for a variety of reasons, from having to calculate certain field values to obnoxious things like int pointers (which, yes, is now moot with the use of `ptr`). It also follows the pattern of "getting" an object rather than directly referencing is from somewhere.

While *here* it's not strictly needed since no logic needs to happen, I'd rather see the function present anyway for the sake of future changes that might require actual computation.

Reply:
@jarrpa 
An established pattern does not mean it is a good pattern. In addition, I don't agree with the statement. We have other controllers inside ocs-operator that do not use this approach. Instead, they use a `templates` module with global vars for all of the static stuff. 


Regarding the `getting` part, from a design perspective it collides with the design behind a reconciliation process. 
With reconciliation, you want to set the desired state of an existing object, not get a new object. 
I can point you to more then one example  where I saw the following pattern 
```
field_i_need_to_preserve0 := oldObject.field0
field_i_need_to_preserve1 := oldObject.field1
field_i_need_to_preserve2 := oldObject.field2
newObj := GetNewObject()
new Obj.DeepCopyInto(oldObj)
oldObject.field0 = field_i_need_to_preserve0
oldObject.field1 = field_i_need_to_preserve1
oldObject.field2 = field_i_need_to_preserve2
```

It shows that the person writing the code put more emphasis on having a "clean" `GetNewObject` than actually solving the problem at hand.

My point is that a blanket statement like this is the pattern we were using is not a good indication id the code fit the needs in this specific instance.